pkg/api/validate/validators.go

```diff
@@ -3,171 +3,180 @@ package validate
+				errs = append(errs, fmt.Errorf("invalid %s[%q]", path, name))
+			}
+		}
+	}
 
-	errs = append(errs, validateFQDN(p, location)...)
-	// we can disregard any error below because we are already going to fail
-	// validation if VnetCIDR does not parse correctly.
-	_, vnet, _ := net.ParseCIDR(p.NetworkProfile.VnetCIDR)
-	errs = append(errs, validateAgentPoolProfiles(p.AgentPoolProfiles, vnet)...)
-	errs = append(errs, validateAuthProfile(&p.AuthProfile)...)
 	return
 }
 
-func validateAuthProfile(ap *api.AuthProfile) (errs []error) {
-	if ap == nil {
-		errs = append(errs, fmt.Errorf("properties.authProfile cannot be nil"))
+func validateRouterProfile(path string, rp *api.RouterProfile, location string, externalOnly bool) (errs []error) {
+	if rp == nil {
+		errs = append(errs, fmt.Errorf("%s cannot be nil", path))
 		return
 	}
 
-	if len(ap.IdentityProviders) != 1 {
-		errs = append(errs, fmt.Errorf("invalid properties.authProfile.identityProviders length"))
+	if !rxRouterProfileName.MatchString(rp.Name) {
+		errs = append(errs, fmt.Errorf("invalid %s", path))
 	}
-	//check supported identity providers
-	for _, ip := range ap.IdentityProviders {
-		switch provider := ip.Provider.(type) {
-		case (*api.AADIdentityProvider):
-			if ip.Name != "Azure AD" {
-				errs = append(errs, fmt.Errorf("invalid properties.authProfile.identityProviders name"))
-			}
-			if provider.Secret == "" {
-				errs = append(errs, fmt.Errorf("invalid properties.authProfile.AADIdentityProvider secret %q", provider.Secret))
-			}
-			if provider.ClientID == "" {
-				errs = append(errs, fmt.Errorf("invalid properties.authProfile.AADIdentityProvider clientId %q", provider.ClientID))
-			}
-			if provider.TenantID == "" {
-				errs = append(errs, fmt.Errorf("invalid properties.authProfile.AADIdentityProvider tenantId %q", provider.TenantID))
-			}
-			if provider.CustomerAdminGroupID != nil && !isValidUUID(*provider.CustomerAdminGroupID) {
-				errs = append(errs, fmt.Errorf("invalid properties.authProfile.AADIdentityProvider customerAdminGroupId %q", *provider.CustomerAdminGroupID))
-			}
+
+	// TODO: consider ensuring that PublicSubdomain is of the form
+	// <string>.<location>.azmosa.io
+	if rp.PublicSubdomain != "" && !isValidHostname(rp.PublicSubdomain) {
+		errs = append(errs, fmt.Errorf("invalid %s.publicSubdomain %q", path, rp.PublicSubdomain))
+	}
+
+	if !externalOnly {
+		if rp.FQDN != "" && !isValidCloudAppHostname(rp.FQDN, location) {
+			errs = append(errs, fmt.Errorf("invalid %s.fqdn %q", path, rp.FQDN))
 		}
+
+		errs = append(errs, validateCertProfile(path+".routerCertProfile", &rp.RouterCertProfile)...)
 	}
+
 	return
 }
 
-func isValidUUID(u string) bool {
-	_, err := uuid.Parse(u)
-	return err == nil
+func validateCertProfile(path string, cp *api.CertProfile) (errs []error) {
+	if cp == nil {
+		errs = append(errs, fmt.Errorf("%s cannot be nil", path))
+		return
+	}
+
+	if !rxKeyVaultSecretURL.MatchString(cp.KeyVaultSecretURL) {
```

Comment:
This does not exist when cluster config is being validated. It being done here:
https://github.com/openshift/openshift-azure/blob/fe0e7390a166187c192c50db765b29af3126a746/cmd/startup/startup.go#L119

So we either need to remove this configurable from external API (not sure why it got there in the first place) or remove validation. 

As per current implementation customer is not providing this.

Reply:
This somehow looks shady :/

Reply:
+ we create vault in fakeRP post validation. so even if we would want to add this before we cant at the moment.

Reply:
The sentiment is right, but a couple of details:
* certs in the vault are available in the internal model once EnrichCSFromVault has been called to pull them
* certs in the internal model are not serialised/persisted
* CertProfile structs do not exist in the external API, at least right now

I'll fix up the ordering problem - thanks for spotting it